snap/snapcraft.yaml

```diff
@@ -64,17 +64,222 @@ parts:
+                          $CRAFT_STAGE/usr/lib/$CRAFT_ARCH_TRIPLET_BUILD_FOR/girepository-1.0\
+                          ${GI_TYPELIB_PATH:+:$GI_TYPELIB_PATH}"
+
+  ninja:
+    plugin: nil
+    source: https://github.com/ninja-build/ninja.git
+    source-tag: 'v1.12.1'
+    source-depth: 1
+    override-build: |
+      rm -rf build
+      rm -f ninja
+      rm -f ninja_bootstrap
+      sed -i 's_^#!/usr/bin/env python$_#!/usr/bin/env python3_g' configure.py
+      ./configure.py --bootstrap
+      mv ninja ninja_bootstrap
+      rm -rf build
+      ./ninja_bootstrap
+      rm -f ninja_bootstrap
+      mkdir -p $CRAFT_PART_INSTALL/usr/bin
+      mv ninja $CRAFT_PART_INSTALL/usr/bin/
+    build-environment: *buildenv
+    build-packages:
+      - python3
+    prime:
+      - -*
+
+  meson-deps:
+    after: [ ninja ]
+    plugin: nil
+    source: https://github.com/mesonbuild/meson.git
+    source-tag: '1.6.0'
+    source-depth: 1
+    override-build: |
+      python3 -m pip install --break-system-packages .
+      mkdir -p $CRAFT_PART_INSTALL/usr/lib/python3/dist-packages
+      rm -rf $CRAFT_PART_INSTALL/usr/lib/python3/dist-packages/meson*
+      python3 -m pip install --target=$CRAFT_PART_INSTALL/usr --break-system-packages .
+      mv $CRAFT_PART_INSTALL/usr/meson* $CRAFT_PART_INSTALL/usr/lib/python3/dist-packages/
+      sed -i "s%^#!/usr/bin/python3$%#!/usr/bin/env python3%g" /usr/local/bin/meson
+      sed -i "s%^#!/usr/bin/python3$%#!/usr/bin/env python3%g" $CRAFT_PART_INSTALL/usr/bin/meson
+    build-environment: *buildenv
+    build-packages:
+      - python3-pip
+    prime:
+      - -*
+
+  pixman:
+    after: [meson-deps]
+    source: https://gitlab.freedesktop.org/pixman/pixman.git
+    source-tag: 'pixman-0.44.2'
+    source-depth: 1
+    plugin: meson
+    meson-parameters:
+      - --prefix=/usr
+      - -Doptimization=3
+      - -Dgtk=disabled
+      - -Dtests=disabled
+      - -Ddemos=disabled
+    build-attributes: [enable-patchelf]
+    build-environment: *buildenv
+    prime:
+      - usr/lib/*/libpix*.so.*
+
+  cairo:
+    after: [pixman, meson-deps]
+    source: https://gitlab.freedesktop.org/cairo/cairo.git
+    source-tag: '1.18.4'
+    source-depth: 1
+    plugin: meson
+    meson-parameters:
+      - --prefix=/usr
+      - -Doptimization=3
+      - -Dxlib=enabled
+      - -Dpng=enabled
+      - -Dxcb=enabled
+      - -Dtee=enabled
+      - -Dzlib=enabled
+      - -Dtests=disabled
+    build-attributes: [enable-patchelf]
+    build-environment: *buildenv
+    build-packages:
+      - libfontconfig1-dev
+      - libfreetype-dev
+      - libx11-dev
+      - libxext-dev
+      - libxcb1-dev
+      - libxcb-render0-dev
+      - libxcb-shm0-dev
+      - libsm-dev
+      - zlib1g-dev
+      - liblzo2-dev
+      - libxml2-dev
+    prime:
+      - usr/lib/*/libcairo*.so.*
+
+  gtk4:
+    after: [meson-deps, cairo]
+    source: https://gitlab.gnome.org/GNOME/gtk.git
+    source-tag: '4.18.2'
```

Comment:
This is available on the `candidate` channel of the gnome-sdk snap: https://github.com/ubuntu/gnome-sdk/commit/50b8d157f3bf721b53ae186ade7ebcf1a3bcaf04